lib/rooms.js

```diff
@@ -324,25 +317,23 @@ Rooms.prototype.write = function write(data) {
  */
 
 Rooms.prototype.broadcast = function broadcast(data, method) {
-
   if ('function' === typeof data[data.length - 1]) {
     throw new RoomsError('Callbacks are not supported when broadcasting');
   }
 
   if (this.id) {
-
-    // We need to make sure this is a valid spark and not a destroyed one,
-    // destroyed spark are no longer part of the connection list
-    var conn = this.connections[this.id];
-
-    // If this is not a valid ctx we just ignore the write request
-    if (!conn) return this.reset();
-
+    // Ignore the broadcast request if the spark is closed
+    if (!this.primus.spark(this.id)) return this.reset();
```

Comment:
I've tried with `this.ctx.readyState === this.primus.Spark.CLOSED` here but unfortunately it doesn't work with `primus-multiplex`.